### ISSUE TITLE
Slick evolutions with compile-time DI

### ISSUE THREAD

**Issue by iref** [playframework/play-slick#269]
I tried to migrate my Play/Slick side-project last weekend and I ran into problem with evolutions.
I use compile-time dependency injection to wire different app components together. In current version of play-slick-evolutions (1.0.0), there isn't easy way to create `DBApi` implementation by hand and wire it in compile time, because `DBApiAdapter` is package private.

So I was wondering if it would be possible to add `SlickEvolutionsComponents` trait, that might look like this:

```scala
trait SlickEvolutionsComponents {
  def slickApi: SlickApi

  lazy val dbApi: DBApi = new DBApiAdapter(api)
}
```

Plugin users would just stack this trait with `EvolutionsComponents` and `SlickComponents` trait.

**Comment by jroper** [comment]
Yes, we should do that, **and** we should make DBApiAdapter public, or expose some public factory method for it - since our approach for DI is that everything can be manually created and wired (the components traits must always be optional).

**Comment by iref** [comment]
I added trait in my fork iref/play-slick@af7a6f77e4a9584b56fb48e72d3638a2ac025878 .

I can also move `DBApiAdapter` object (and class?) out from internal package and expose factory method in `DBApiAdapter` object. I would keep `DatabaseAdapter` class package private in `internals ` package. This way only `DBApiAdapter.apply(slickApi)` method would be public and implementation details are still hidden.

**Comment by jroper** [comment]
That would be fine.  I don't actually like the name `DBApiAdapter`, I'd prefer `SlickDBApi.apply(slickApi)` as the public method.

**Comment by iref** [comment]
Sound good. I agree, `DBApiAdapter` is probably too generic name.

**Comment by andrewlague** [comment]
Any idea when this fix will be released (1.0.1, I assume?) I'm having trouble using it through a source SBT dependency.
